docs/user-guide/SPIN.rst

```diff
@@ -0,0 +1,181 @@
+##################################################
+Unlike DPO and PPO, SPIN was designed to be run on foundational (base) models, that is, models which have only been trained on autoregressive language prediction tasks and not on instruction following tasks.
+However, you can also run SPIN on models which have been SFTed on instruction-based datasets as well, similar to DPO/PPO. Either type of model will work well with SPIN. If you would like to start with a supervised fine tuned model instead of a base model, please see our full guide on how to perform SFT on a Megatron GPT model :ref:`SFT guide <sft>`.
+
+SPIN Model Training
+###################
+
+SPIN training uses the exact same dataset formatting and files as the NeMo-Aligner SFT trainer. Please see the data formatting section of SFT to understand the data format necessary for SPIN :ref:`SFT guide <sft>`
+
+Once your data is processed into the correct format you are ready to begin SPIN training. You must start with a pretrained or SFT trained model. For this section we will use the SFT model trained in the previous step to train the SPIN model.
+For the purposes of the following sections, we'll assuming your training jsonl file is located in ``/path/to/train_spin_format.jsonl`` and your validation jsonl file is located in ``/path/to/valid_spin_format.jsonl``.
+
+For the below parameters, the ``model.spin.ref_policy_kl_penalty`` corresponds to the beta parameter in the SPIN paper, and ``trainer.spin.max_iterations`` corresponds to T (with ``trainer.spin.max_epochs`` per iteration)
+
+.. tab-set::
+
+    .. tab-item:: Terminal
+        :sync: key3
+
+         To run SPIN model training on the terminal directly
+
+         .. code-block:: bash 
+
+            export GPFS="/path/to/nemo-aligner-repo"
+            export TRAIN_DATA_PATH="/path/to/train_spin_format.jsonl"
+            export VALID_DATA_PATH="/path/to/valid_spin_format.jsonl"
+
+            python -u ${GPFS}/examples/nlp/gpt/train_gpt_spin.py \
+               trainer.num_nodes=1 \
+               trainer.devices=8 \
+               ++model.mcore_gpt=true \
+               ++model.micro_batch_size=1 \
+               ++model.global_batch_size=64 \
+               pretrained_checkpoint.restore_from_path=/path/to/megatron_gpt_sft.nemo \
+               "model.data.train_ds.file_path=${TRAIN_DATA_PATH}" \
+               "model.data.validation_ds.file_path=${VALID_DATA_PATH}" \
+               exp_manager.create_wandb_logger=false \
+               exp_manager.wandb_logger_kwargs.project=spin_training \
+               exp_manager.wandb_logger_kwargs.name=spin_training \
+               exp_manager.explicit_log_dir=/results \
+               ++trainer.spin.max_iterations=1 \
+               ++trainer.spin.max_epochs=1 \
+               ++model.spin.ref_policy_kl_penalty=0.1
+               ++model.spin.length_params.max_length=2048 \
+               ++model.data.train_ds.max_seq_length=4096 \
```

Comment:
Couple of comments:
1. Small syntax error probably due to moving lines around => the line with `ref_policy_kl_penalty` should end with a `\` and the last line shouldn't
2. Better remove the `++` in front of all of these options since they are in our config so the `++` is useless (otherwise it's confusing as to why we need `++`, and also we may not notice that our tutorials are broken if a variable is renamed)

Reply:
Fixed, thanks